src/main/java/uk/gov/hmcts/darts/audio/controller/AudioController.java

```diff
@@ -61,11 +78,28 @@ public ResponseEntity<byte[]> preview(Integer mediaId, String httpRangeList) {
         return StreamingResponseEntityUtil.createResponseEntity(audioMediaFile, httpRangeList);
     }
 
-    @Override
+
+    @RequestMapping(
+        method = RequestMethod.GET,
+        value = "/audio/preview/{media_id}",
+        produces = {"text/event-stream", "application/json+problem"}
+    )
+    @SneakyThrows
     @SecurityRequirement(name = SECURITY_SCHEMES_BEARER_AUTH)
-    public ResponseEntity<Void> addAudio(MultipartFile file, AddAudioMetadataRequest metadata) {
-        audioService.addAudio(file, metadata);
-        return new ResponseEntity<>(HttpStatus.OK);
+    @Authorisation(contextId = MEDIA_ID,
+        securityRoles = {JUDGE, REQUESTER, APPROVER, TRANSCRIBER, TRANSLATION_QA, RCJ_APPEALS},
+        globalAccessSecurityRoles = {JUDGE})
+    public Flux<ResponseEntity<byte[]>> previewAlternative(
+        @Parameter(name = "media_id", description = "Internal identifier for media", required = true, in = ParameterIn.PATH)
+        @PathVariable("media_id") Integer mediaId,
+        @Parameter(name = "range", description = "Range header, required for streaming audio.", in = ParameterIn.HEADER)
+        @RequestHeader(value = "range", required = false) String range
+    ) {
+        Flux<ResponseEntity<byte[]>> work = audioService.getAudioPreviewFlux(mediaId, range);
+
+        Flux<ResponseEntity<byte[]>> heartBeat = Flux.interval(Duration.ofSeconds(5))
+            .map(sequence -> new ResponseEntity<byte[]>(HttpStatus.CONTINUE)).takeUntilOther(work);
```

Comment:
Does this break the HTTP semantics of continue. HTTP is request response and continue has a specific meaning. This seems like a bespoke scenario for the client connection to stay open?

Reply:
Looking at the definition of it I think matches our usage - "The HTTP 100 Continue informational status response code indicates that everything so far is OK and that the client should continue with the request or ignore it if it is already finished."